src/main/java/seedu/address/ui/EventSpaceBackground.java

```diff
@@ -0,0 +1,119 @@
+/**
+ * The UI component that represents the background grid for an event space, displaying time intervals.
+ */
+public class EventSpaceBackground extends UiPart<Region> {
+    private static final String FXML = "EventSpaceBackground.fxml";
+    private static final LocalTime DEFAULT_CALENDAR_START_TIME = LocalTime.of(8, 0);
+    private static final LocalTime DEFAULT_CALENDAR_END_TIME = LocalTime.of(18, 0);
+    private static final int NUMBER_OF_HOURS_IN_ONE_DAY = 24;
+    private static final int MAXIMUM_DISPLAY_HOUR = 23;
+    private static final int NUMBER_OF_MINUTES_IN_AN_HOUR = 60;
+
+    private LocalTime calendarStartTime = DEFAULT_CALENDAR_START_TIME;
+    private LocalTime calendarEndTime = DEFAULT_CALENDAR_END_TIME;
+
+    private ReadOnlyCalendar calendar;
+    private ObservableList<Event> eventList;
+    private ObservableList<EventSpaceBackgroundColumn> columns;
+
+    @FXML
+    private HBox backgroundGrid;
+
+    /**
+     * Constructs an EventSpaceBackground with the given ReadOnlyCalendar.
+     *
+     * @param calendar The ReadOnlyCalendar to be displayed in the background.
+     */
+    public EventSpaceBackground(ReadOnlyCalendar calendar) {
+        super(FXML);
+        this.calendar = calendar;
+        eventList = calendar.getCurrentWeekEventList();
+        columns = FXCollections.<EventSpaceBackgroundColumn>observableArrayList();
+
+        fillBackground();
+        updateStartAndEnd();
+        showRelevantBackground();
+        addListenerToEventList();
+    }
+
+
+    /**
+     * Adds a listener to the event list to update the background when events change.
+     */
+    private void addListenerToEventList() {
+        eventList.addListener((ListChangeListener<Event>) c -> {
+            updateStartAndEnd();
+            showRelevantBackground();
+        });
+    }
+
+    /**
+     * Fills the background with time intervals.
+     */
+    public void fillBackground() {
+        Stream.<LocalTime>iterate(LocalTime.MIDNIGHT, time -> time.plusHours(1))
+                .limit(NUMBER_OF_HOURS_IN_ONE_DAY)
+                .map(EventSpaceBackgroundColumn::new)
+                .forEachOrdered(column -> columns.add(column));
+        columns.forEach(column -> backgroundGrid.getChildren().add(column.getRoot()));
+    }
+
+    /**
+     * Updates the start and end times based on the events in the calendar.
+     */
+    private void updateStartAndEnd() {
+        LocalTime newStartTime = calendar.getEarliestEventStartTimeInCurrentWeek()
+                .map(time -> {
+                    return time.minusMinutes(time.getMinute());
+                })
+                .orElse(calendarStartTime);
+        LocalTime newEndTime = calendar.getLatestEventEndTimeInCurrentWeek()
+                .map(time -> {
+                    if (time.getMinute() == 0) {
+                        return time;
+                    }
+
+                    if (time.getHour() == MAXIMUM_DISPLAY_HOUR) {
+                        return time;
```

Comment:
Similar responses for 2 if statements can be condensed into 1 after storing conditions as variables.